apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClientImpl.SingleChannel.cs

```diff
@@ -279,7 +279,11 @@ private void StopStreamingPull()
                     _registerTaskFn(_pull.WriteCompleteAsync());
                 }
                 catch { }
-                _pull = null;
+                finally
+                {
+                    _pull.Dispose();
```

Comment:
This one need to be registered as well, and as a continuation of the WriteCompleteAsync.
Else Dispose will likely be called before WriteCompleteAsync and WriteCompleteAsync will fail.

Reply:
Right. 
I had been registering it all this while (for deadlock issue and even today) and just before raising the PR, I changed it. Reverted now. PTAL.